### PR TITLE
adding an alert about dep-check, plus some refactoring

### PR DESCRIPTION
**Pull request by kelset**
# Summary

Started from this [twitter thread](https://twitter.com/lbentosilva/status/1522457986077179911) with @lucasbento & @pvinis, I decided to add in the callout section the fact that dep-check is a thing and that should help you :)

Copy is still a WIP tbh, I'm going to ask @afoxman and @tido64 to give their 👍 (on top of Lucas' and Pavlos', ofc!)

While doing so I realised that I wanted to do a bit more 😇 so I did a bit of refactoring so that on all releases we show some information that are not strictly version-bound (see screenshots for more). This also sets the stage for extracting from the links part the changelog bit so that we can also always parse up the RN versions changelog to show in the callout... but I left that part for some future work (potentially some good first issue material?).

Let me know what you think! :)

### Before/after for when there are no useful links
Screenshot 2022-05-06 at 12 46 02

### Before/after for when there are useful links
Screenshot 2022-05-06 at 12 46 24

## What are the steps to reproduce?

Just run the website on the branch :)

## Checklist

- [X] I tested this thoroughly
- [ ] I added the documentation in `README.md` (if needed)

### CODE REVIEW

File: src/components/common/DepCheckAlert.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import Markdown from './Markdown'
+
+const DepCheckAlert = () => (
+  <>
+    <Markdown>
+      You can use the following command to kick off the upgrade: `npx
+      rnx-dep-check --set-version [major.minor]`.
```
Comment (tido64) [inline src/components/common/DepCheckAlert.js:8]:
```suggestion
      @rnx-kit/dep-check --set-version [major.minor]`.
```

**Review by tido64 — APPROVED** [review]

File: src/components/common/DepCheckAlert.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <Markdown>
+      You can use the following command to kick off the upgrade: `npx
+      @rnx-kit/dep-check --set-version [major.minor]`.
+    </Markdown>
+    <br />
+    <Markdown>
+      `dep-check` is a tool by Microsoft that helps your dependencies
+      management, aligning the main RN libraries on the version of RN used (find
+      out more in the [dedicated
+      documentation](https://microsoft.github.io/rnx-kit/docs/guides/dependency-management)).
```
Comment (afoxman) [inline src/components/common/DepCheckAlert.js:15]:
```suggestion
      `dep-check` is a **community** tool from Microsoft that automates dependency
      management. It knows which RN package versions work well together and are 
      compatible with your specific version of RN. It uses that knowledge to align dependencies,
      keeping your app healthy and up-to-date. Find out more [here](https://microsoft.github.io/rnx-kit/docs/guides/dependency-management)).
```

**Comment by cortinico** [comment]
Great stuff :)

File: src/components/common/DepCheckAlert.js (hunk 2 above)
Comment (leotm) [reply src/components/common/DepCheckAlert.js:15]: could even coin the term [community-first](https://microsoft.github.io/rnx-kit/blog/react-native-developer-tools#community-first) (;

enjoyed the 1st blogpost, the 4th was with me wednesday

Comment (leotm) [reply src/components/common/DepCheckAlert.js:15]:
```diff
- keeping your app healthy and up-to-date. ... Find out more [here] ...
+ keeping your app healthy and up-to-date, far from [the shores](https://www.microsoft.com/p/doom-1993/9plzphbnhtmf) of [dependency hell](https://en.wikipedia.org/wiki/Dependency_hell). Find out more [here] ...
```

the [episode 2](https://doomwiki.org/wiki/The_Shores_of_Hell) mention was meant to be there

microsoft aquiring github, then the doom franchise... nearly 30 yrs multi-platform from support ms-dos to win11...

who knows what lies ahead :godmode:

**Comment by kelset** [comment]
thanks y'all for the feedback, here's the new copy:
Screenshot 2022-05-09 at 14 17 32

**Review by lucasbento — APPROVED** [review]

**Pull request merged**